kotlin/workflow-testing/src/main/java/com/squareup/workflow/testing/RenderTester.kt

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2019 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.workflow.testing
+
+import com.squareup.workflow.StatefulWorkflow
+import com.squareup.workflow.Worker
+import com.squareup.workflow.Workflow
+import com.squareup.workflow.WorkflowAction
+import kotlin.reflect.KClass
+
+/**
+ * Create a [RenderTester] to unit test an individual render pass of this workflow.
+ *
+ * See [RenderTester] for usage documentation.
+ */
+@Suppress("UNCHECKED_CAST")
+fun <PropsT, OutputT : Any, RenderingT> Workflow<PropsT, OutputT, RenderingT>.renderTester(
+  props: PropsT
+): RenderTester<PropsT, *, OutputT, RenderingT> {
+  val statefulWorkflow = asStatefulWorkflow() as StatefulWorkflow<PropsT, Any?, OutputT, RenderingT>
+  return statefulWorkflow.renderTester(
+      props = props,
+      initialState = statefulWorkflow.initialState(props, null)
+  ) as RenderTester<PropsT, Nothing, OutputT, RenderingT>
+}
+
+/**
+ * Create a [RenderTester] to unit test an individual render pass of this workflow.
+ *
+ * See [RenderTester] for usage documentation.
+ */
+// @formatter:off
+fun <PropsT, StateT, OutputT : Any, RenderingT>
+    StatefulWorkflow<PropsT, StateT, OutputT, RenderingT>.renderTester(
+      props: PropsT,
+      initialState: StateT
+    ): RenderTester<PropsT, StateT, OutputT, RenderingT> =
+// @formatter:on
+  RealRenderTester(this, props, initialState)
+
+/**
+ * The props must be specified, the initial state may be specified, and then all child workflows
+ * and workers that are expected to run must be specified with [expectWorkflow] and [expectWorker]
+ * calls. Then call [render] and perform any assertions on the rendering. An event may also be sent
```

Comment:
You're implying here that, in addition to being able to send an event to a rendering, I can also simulate a child worker or workflow emitting output. Seems like that should be said out loud. Bonus points for providing an example.

Reply:
Added a bunch more examples, and updated this paragraph to mention that `expect*` can be used to configure outputs.